From fd91646fbae9b0f7ce90fb5d7dda34e8f13856fe Mon Sep 17 00:00:00 2001
From: Tim Coninx <tim.coninx@gmail.com>
Date: Wed, 12 Aug 2026 16:05:08 +0200
Subject: [PATCH 1/3] Check for deprecated attributes in GnomeDesktop which
 have been removed in newer versions. This prevents AttributeError when trying
 to access them.

---
 lutris/util/display.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lutris/util/display.py b/lutris/util/display.py
index d8f96861f0..6b0e243d95 100644
--- a/lutris/util/display.py
+++ b/lutris/util/display.py
@@ -14,9 +14,11 @@
 try:
     gi.require_version("GnomeDesktop", "3.0")
     from gi.repository import GnomeDesktop  # type: ignore
+    # check for AttributeError on newer systems on which the deprecated attributes have been removed
+    GnomeDesktop.RRScreen  # type: ignore
 
     LIB_GNOME_DESKTOP_AVAILABLE = True
-except ValueError:
+except (ValueError, AttributeError):
     LIB_GNOME_DESKTOP_AVAILABLE = False
     GnomeDesktop = None
 

From 68197a0686526df1c117d4e2506d5fd8af4cb2ae Mon Sep 17 00:00:00 2001
From: Tim Coninx <tim.coninx@gmail.com>
Date: Wed, 12 Aug 2026 16:15:02 +0200
Subject: [PATCH 2/3] Comment formatting

---
 lutris/util/display.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lutris/util/display.py b/lutris/util/display.py
index 6b0e243d95..f468e89a8c 100644
--- a/lutris/util/display.py
+++ b/lutris/util/display.py
@@ -14,7 +14,8 @@
 try:
     gi.require_version("GnomeDesktop", "3.0")
     from gi.repository import GnomeDesktop  # type: ignore
-    # check for AttributeError on newer systems on which the deprecated attributes have been removed
+    # check for AttributeError on newer systems on which the
+    # deprecated attributes have been removed
     GnomeDesktop.RRScreen  # type: ignore
 
     LIB_GNOME_DESKTOP_AVAILABLE = True

From 27ba1c90242597e72380fdc505249581fb8ddb44 Mon Sep 17 00:00:00 2001
From: Tim Coninx <tim.coninx@gmail.com>
Date: Wed, 12 Aug 2026 16:24:06 +0200
Subject: [PATCH 3/3] Ruff formatting

---
 lutris/util/display.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lutris/util/display.py b/lutris/util/display.py
index f468e89a8c..0a47b3cdae 100644
--- a/lutris/util/display.py
+++ b/lutris/util/display.py
@@ -14,6 +14,7 @@
 try:
     gi.require_version("GnomeDesktop", "3.0")
     from gi.repository import GnomeDesktop  # type: ignore
+
     # check for AttributeError on newer systems on which the
     # deprecated attributes have been removed
     GnomeDesktop.RRScreen  # type: ignore
